library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionListFragment.kt

```diff
@@ -107,7 +131,51 @@ internal class TransactionListFragment :
         TransactionActivity.start(requireActivity(), transactionId)
     }
 
+    private fun exportTransactions() {
+        viewModel.getAllTransactions { transactions ->
```

Comment:
This whole function is hard to read, especially when we get to part after `else`. 
Let's change it to plain simple functions, like `viewModel.getAllTransactions()` just returns a `List` and `createExportFile` just returns file.  
We need to try keep library code as simple as possible, so everyone is able to understand it fast.

Reply:
agreed, my bad. I have refactored to make it simpler please do take another look at this function.